sql/core/src/test/scala/org/apache/spark/sql/connector/DataSourceV2Suite.scala

```diff
@@ -966,6 +966,22 @@ class DataSourceV2Suite extends QueryTest with SharedSparkSession with AdaptiveS
       )
     }
   }
+
+  test("SPARK-47463: Pushed down v2 filter with if expression") {
+    withTempView("t1") {
+      spark.read.format(classOf[AdvancedDataSourceV2WithV2Filter].getName).load()
+        .createTempView("t1")
+      val df1 = sql(
+        s"""
+           |select * from
+           |(select if(i = 1, i, 0) as c from t1) t
+           |where t.c > 0
```

Comment:
I think `where if(i = 1, i, 0) > 0` is a valid SQL? BTW, please upper case the SQL keywords in the SQL statement.

Reply:
> I think `where if(i = 1, i, 0) > 0` is a valid SQL? BTW, please upper case the SQL keywords in the SQL statement.

Indeed, thank you. I have changed.